src/alfasim_score/converter/alfacase/production_operation.py

```diff
@@ -117,14 +123,38 @@ def configure_well_initial_conditions(self, alfacase: CaseDescription) -> None:
             ),
             temperatures=self.create_well_initial_temperatures(
                 Scalar(formation_data["temperatures"][0], TEMPERATURE_UNIT),
-                operation_data["flow_initial_temperature"],
+                self.general_data["flow_initial_temperature"],
+            ),
+        )
+
+    def configure_physics(self, alfacase: CaseDescription) -> None:
+        """Configure the description for the physics data."""
+        super().configure_physics(alfacase)
+        no_inlet_water = np.isclose(self.general_data["water_flow_rate"].GetValue(), 0.0)
```

Comment:
The same way we have `self.has_gas_lift()` we could have a `self.has_water()`.
It could check if `self.general_data["water_flow_rate"]` is zero (or close) for production operation and `fluid_type` for injection operation.

Reply:
I created new methods to encapsulate these checks.